server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -391,13 +391,20 @@ public Text readOptionalText() throws IOException {
         if (length == -1) {
             return null;
         }
-        return new Text(readBytesReference(length));
+        var byteBuffs = BytesReference.toByteBuffers(readBytesReference(length));
```

Comment:
Why is this implementation different from readText below? The implementation below makes more sense (it's easier to understand and probably more efficient)